test/e2e/network/netpol/policies.go

```diff
@@ -18,564 +18,259 @@ package netpol
 
 import (
 	"fmt"
-
 	"k8s.io/apimachinery/pkg/util/intstr"
 
 	v1 "k8s.io/api/core/v1"
 	networkingv1 "k8s.io/api/networking/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type SetFunc func(policy *networkingv1.NetworkPolicy)
+
+func CreateNetworkPolicy(fn ...SetFunc) *networkingv1.NetworkPolicy {
+	policy := &networkingv1.NetworkPolicy{}
+	for _, f := range fn {
+		f(policy)
+	}
+	return policy
+}
+
+func SetObjectMetaName(name string) SetFunc {
+	return func(policy *networkingv1.NetworkPolicy) {
+		policy.ObjectMeta.Name = name
+	}
+}
+
+func SetSpecPodSelector(targetSelector metav1.LabelSelector) SetFunc {
+	return func(policy *networkingv1.NetworkPolicy) {
+		policy.Spec.PodSelector = targetSelector
+	}
+}
+
+func SetSpecPodSelectorMatchLabels(targetLabels map[string]string) SetFunc {
+	return func(policy *networkingv1.NetworkPolicy) {
+		policy.Spec.PodSelector = metav1.LabelSelector{
+			MatchLabels: targetLabels,
+		}
+	}
+}
+
+func SetSpecIngressRules(rules ...networkingv1.NetworkPolicyIngressRule) SetFunc {
+	return func(policy *networkingv1.NetworkPolicy) {
+		if policy.Spec.Ingress == nil {
+			policy.Spec.Ingress = []networkingv1.NetworkPolicyIngressRule{}
+			policy.Spec.PolicyTypes = append(policy.Spec.PolicyTypes, networkingv1.PolicyTypeIngress)
+		}
+		for _, rule := range rules {
+			policy.Spec.Ingress = append(policy.Spec.Ingress, rule)
+		}
+	}
+}
+
+func SetSpecEgressRules(rules ...networkingv1.NetworkPolicyEgressRule) SetFunc {
+	return func(policy *networkingv1.NetworkPolicy) {
+		if policy.Spec.Egress == nil {
+			policy.Spec.Egress = []networkingv1.NetworkPolicyEgressRule{}
+			policy.Spec.PolicyTypes = append(policy.Spec.PolicyTypes, networkingv1.PolicyTypeEgress)
+		}
+		for _, rule := range rules {
+			policy.Spec.Egress = append(policy.Spec.Egress, rule)
+		}
+	}
+}
+
 // GetDenyIngress returns a default deny ingress policy.
 func GetDenyIngress(name string) *networkingv1.NetworkPolicy {
-	return &networkingv1.NetworkPolicy{
```

Comment:
It's a good improvement in the API but less intuitive and harder to read than before, I wonder if some comments with the spec boilerplate at the beginning of each function could become useful.

Other than that LGTM.